pkg/reconciler/pipelinerun/resources/conditionresolution.go

```diff
@@ -174,13 +174,22 @@ func (rcc *ResolvedConditionCheck) ToTaskResourceBindings() []v1alpha1.TaskResou
 	var trb []v1alpha1.TaskResourceBinding
 
 	for name, r := range rcc.ResolvedResources {
-		trb = append(trb, v1alpha1.TaskResourceBinding{
+		tr := v1alpha1.TaskResourceBinding{
 			Name: name,
-			ResourceRef: v1alpha1.PipelineResourceRef{
+		}
+		if r.SelfLink != "" {
```

Comment:
hey @pritidesai, somehow I didn't notice this when I looked before: what's the significance of `selfLink` here? it seems like its being used to decide if a resource should be embedded or not? 🤔

Reply:
> hey @pritidesai, somehow I didn't notice this when I looked before: what's the significance of `selfLink` here? it seems like its being used to decide if a resource should be embedded or not? 🤔

Hey @bobcatfish yes `selfLink` is the deciding factor here and most reliable, if resource is specified through reference (`resourceRef`), `selfLink` will never be empty  and set to something like `/apis/tekton.dev/pipelineresources/my-git-resource` since `my-git-resource` is already created and can be referred to through this link. This was the best choice, as in `inputs` being type of `map[string]*v1alpha1.PipelineResource` in [input_output_steps.go line 56](https://github.com/tektoncd/pipeline/blob/master/pkg/reconciler/pipelinerun/resources/input_output_steps.go#L56) always had `name` set to the resource name and `PipelineResource.name` set to the same name as resource name. Hope this helps.